pkg/controller/daemon/daemoncontroller.go

```diff
@@ -716,6 +744,17 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 		}
 	}
 
+	dsNeedsSync = dsc.expectations.SatisfiedExpectations(dsKey)
```

Comment:
Already called above. Although expectations never worked as expected and we should probably get rid of them from this controller too.